plugins/database/src/db_plugin.cpp

```diff
@@ -7089,64 +7196,223 @@ irods::error db_mod_user_op(
             addRErrorMsg( &_ctx.comm()->rError, 0, "acCheckPasswordStrength rule not found" );
         }
 
-
         if ( status2 ) {
             return ERROR( status2, "icatApplyRule failed" );
         }
 
-        icatScramble( decoded );
-
-        if ( i ) {
-            return ERROR( i, "password scramble failed" );
-        }
-        if ( logSQL != 0 ) {
-            log_sql::debug("chlModUser SQL 8");
-        }
+        const auto config_handle{irods::server_properties::instance().map()};
+        const auto& config{config_handle.get_json()};
+        auto store_hashed_password = false;
+        auto store_scrambled_password = true;
+        if (const auto password_storage_mode_iter = config.find(irods::KW_CFG_USER_PASSWORD_STORAGE_MODE);
+            config.end() != password_storage_mode_iter)
         {
-            std::vector<std::string> bindVars;
-            bindVars.push_back( userName2 );
-            bindVars.push_back( zoneName );
-            i = cmlGetStringValueFromSql(
-                    "select R_USER_PASSWORD.user_id from R_USER_PASSWORD, R_USER_MAIN where R_USER_MAIN.user_name=? and R_USER_MAIN.zone_name=? and R_USER_MAIN.user_id = R_USER_PASSWORD.user_id",
-                    userIdStr, MAX_NAME_LEN, bindVars, &icss );
+            const auto& password_storage_mode = password_storage_mode_iter->get_ref<const std::string&>();
+            store_scrambled_password = (password_storage_mode != "hashed");
+            store_hashed_password = (password_storage_mode != "legacy");
         }
-        if ( i != 0 && i != CAT_NO_ROWS_FOUND ) {
-            return ERROR( i, "get user password failed" );
+
+        if (store_hashed_password) {
+            try {
+                // Get the password hashing parameters from the grid configuration table.
+                std::vector<std::string> bindVars{"authentication", "password_hashing_parameters"};
+                constexpr auto max_size_for_params = 2700;
+                std::vector<char> params_str(max_size_for_params + 1);
+                {
+                    const int ec = cmlGetStringValueFromSql(
+                        "select option_value from R_GRID_CONFIGURATION where namespace = ? and option_name = ?",
+                        params_str.data(),
+                        max_size_for_params,
+                        bindVars,
+                        &icss);
+
+                    if (ec < 0) {
+                        const auto msg = fmt::format(
+                            "{}: Failed to get configuration for hashing parameters. Cannot set password.", __func__);
+                        log_db::info(msg);
+                        return ERROR(ec, msg);
```

Comment:
Can drop `const` from `msg` and `std::move(msg)` into `ERROR()`.

Reply:
Done